lib/resources/google_apis.dart

```diff
@@ -0,0 +1,85 @@
+import 'dart:io' show Platform;
+import 'package:acs_upb_mobile/pages/timetable/model/events/uni_event.dart';
+import 'package:flutter/foundation.dart' show kIsWeb;
+import 'package:googleapis_auth/auth_io.dart';
+import 'package:googleapis/calendar/v3.dart';
+
+class GoogleApiHelper {
+  GoogleApiHelper();
+
+  static const List<String> _scopes = [CalendarApi.CalendarScope];
+
+  static List<String> get scopes => _scopes;
+
+  static ClientId get credentials {
+    String _clientIdString;
+
+    if (Platform.isAndroid) {
+      _clientIdString =
+          '611150208061-4ftun8ln4v9hm1mocqs1vqcftaanj8sj.apps.googleusercontent.com';
+    } else if (Platform.isIOS) {
+      _clientIdString =
+          '611150208061-4ftun8ln4v9hm1mocqs1vqcftaanj8sj.apps.googleusercontent.com';
+    } else if (kIsWeb) {
+      _clientIdString =
+          '611150208061-ljqdu5mfmjisdi1h3ics3l2sirvtpljk.apps.googleusercontent.com';
+    } else {
+      _clientIdString =
+          '611150208061-ljqdu5mfmjisdi1h3ics3l2sirvtpljk.apps.googleusercontent.com';
+    }
+    return ClientId(_clientIdString, '');
+  }
+}
+
+enum GoogleCalendarColorNames {
+  undefined,
+  lavender,
+  sage,
+  grape,
+  flamingo,
+  banana,
+  tangerine,
+  peacock,
+  graphite,
+  blueberry,
+  basil,
+  tomato
+}
```

Comment:
Why not just make this a `Map<String, Color>`? It'd let you delete almost all the code below. I don't really see the benefit in defining a custom type for these.

Reply:
I need the index value since that Is the color ID, used by Google Calendar. But I didn't want to write "magic numbers" so I used these official identifiers. One other solution would be a Map<String,int> for names of colors and another one for hex codes. Or should I use a [tuple](https://pub.dev/packages/tuple) for colors and hex codes?

Reply:
I really don't see the need for hex codes. I took a look at the API (hadn't when I made the initial comment) and as far as I understand, it's as simple as "if you want lavender, set the color ID to 1". You can keep the code as is but remove GoogleCalendarColorHexValues altogether since it's not used anywhere, it's just confusing. If you wanna keep a reference of the hex of each color, just put it in a comment e.g.:
```
lavender,  // #7986cb
```